### PR TITLE
chore: add redux-toolkit (rtk) to sql runner

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: N/A
Relates to: #10515 

### Description:

Adds RTK - redux toolkit
Wraps the Sql Runner with `Provider` and a `store`
Handles the selection of an active table and active fields
Ensures that all the fields don't get rendered when a field is selected (through `memo` and `dispatch`, instead of `useState` or a `Context provider`)

Then on the folder structure
- Adds a `store`
- Adds a SQL Runner `slice`
- Adds a `hooks` file that exports typed hooks for dispatching and selecting data

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/store/index.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { configureStore } from '@reduxjs/toolkit';
```
Comment (ZeRego) [inline packages/frontend/src/store/index.ts:1]: We should move all the redux files to `packages/frontend/src/store/features/sqlRunner/store/`. 
Avoid confusion and prevent other devs from using this until we are sure about this tech decision.

File: packages/frontend/src/pages/SqlRunnerNew.tsx [hunk 2]
```diff
@@ -16,63 +18,67 @@ const SqlRunnerNewPage = () => {
...
-            isRightSidebarOpen={isRightSidebarOpen}
-            rightSidebar={<RightSidebar setSidebarOpen={setRightSidebarOpen} />}
-        >
-            <Group
-                align={'stretch'}
-                grow
-                spacing="none"
-                p={0}
-                style={{ flex: 1 }}
+        <Provider store={store}>
```
Comment (ZeRego) [inline packages/frontend/src/pages/SqlRunnerNew.tsx:21]: Can we require the project uuid to be passed here? + remove prop drilling

File: packages/frontend/src/store/features/sqlRunner/sqlRunnerSlice.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            const index = stateFields.indexOf(field);
+
+            if (index === -1) {
+                state.activeFields = [...stateFields, field];
+            } else {
+                state.activeFields = [
+                    ...stateFields.slice(0, index),
+                    ...stateFields.slice(index + 1),
+                ];
+            }
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/frontend/src/store/features/sqlRunner/sqlRunnerSlice.ts:32]: Great, I had no idea that existed...
I think we should actually remove this `toggleActiveField` until we know if these are clickable (table will remain).

Should I rename then `setActiveTable` to `toggleActiveTable` perhaps?

File: packages/frontend/src/pages/SqlRunnerNew.tsx (hunk 2 above)
Comment (joaoviana) [reply packages/frontend/src/pages/SqlRunnerNew.tsx:21]: oooh checking!

File: packages/frontend/src/store/index.ts (hunk 1 above)
Comment (joaoviana) [reply packages/frontend/src/store/index.ts:1]: sounds good!

File: packages/frontend/src/store/features/sqlRunner/sqlRunnerSlice.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ZeRego) [reply packages/frontend/src/store/features/sqlRunner/sqlRunnerSlice.ts:32]: > I think we should actually remove this toggleActiveField until we know if these are clickable (table will remain).

👍

> Should I rename then setActiveTable to toggleActiveTable perhaps?

👍

File: packages/frontend/src/pages/SqlRunnerNew.tsx (hunk 2 above)
Comment (joaoviana) [reply packages/frontend/src/pages/SqlRunnerNew.tsx]: LMK what you think of this approach: https://github.com/lightdash/lightdash/pull/10589/commits/27eb69d1da2595d1fe22f1a5106a12b140b534c2

File: packages/frontend/src/store/index.ts (hunk 1 above)
Comment (joaoviana) [reply packages/frontend/src/store/index.ts]: done here: https://github.com/lightdash/lightdash/pull/10589/commits/51aab283e02b2f42ce75ef5cc17d18d6f9a1f951

File: packages/frontend/src/store/features/sqlRunner/sqlRunnerSlice.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/frontend/src/store/features/sqlRunner/sqlRunnerSlice.ts]: Simplified `TableField` -> [chore: simplify TableField](https://github.com/lightdash/lightdash/pull/10589/commits/ffee69a452de5a964f1dcaffd40412a13cf99c75)

[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/frontend/src/store/features/sqlRunner/sqlRunnerSlice.ts]: Renamed to `toggleActiveTable`: https://github.com/lightdash/lightdash/pull/10589/commits/36a7dea3b9e110d630f162bd1bdbf45ed0a9e53a

**Comment by joaoviana** [comment]
Fixes TS errors: 
```
20.34 src/features/sqlRunner/store/hooks.ts(5,14): error TS2742: The inferred type of 'useAppDispatch' cannot be named without a reference to '@reduxjs/toolkit/node_modules/redux'. This is likely not portable. A type annotation is necessary.
20.34 src/features/sqlRunner/store/hooks.ts(5,14): error TS2742: The inferred type of 'useAppDispatch' cannot be named without a reference to '@reduxjs/toolkit/node_modules/redux-thunk'. This is likely not portable. A type annotation is necessary.
20.34 src/features/sqlRunner/store/index.ts(4,14): error TS2742: The inferred type of 'store' cannot be named without a reference to '@reduxjs/toolkit/node_modules/redux'. This is likely not portable. A type annotation is necessary.
20.34 src/features/sqlRunner/store/index.ts(4,14): error TS2742: The inferred type of 'store' cannot be named without a reference to '@reduxjs/toolkit/node_modules/redux-thunk'. This is likely not portable. A type annotation is necessary.
20.34 src/features/sqlRunner/store/sqlRunnerSlice.ts(32,1): error TS2742: The inferred type of 'default' cannot be named without a reference to '@reduxjs/toolkit/node_modules/redux'. This is likely not portable. A type annotation is necessary.
```

https://github.com/lightdash/lightdash/pull/10589/commits/8407f23801fccfe6cc73ff9e7ad291b8b5f7987d
with solution proposed here: https://github.com/reduxjs/redux-toolkit/issues/3962#issuecomment-1936754958

**Review by ZeRego — APPROVED** [review]

**Pull request merged**